docs(allocator/vec): correct safety comments for `RawVec::append_elements`

overlookmotel opened the pull request:
Follow-on after #10884.

The original safety conditions aren't required any more, because of the change made to `Vec::reserve` in #10884 - `reserve` will panic if `other.len` or `self.len + other.len` exceeds `u32::MAX`. So nothing can go wrong if these conditions aren't satisfied.

Which is lucky, because `append` doesn't do anything to make sure `self.len + other.len <= u32::MAX`!

`append_elements` originally had no safety docs at all (naughty bumpalo!). So add some.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11072?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#11074** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11074?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#11073** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11073?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#11072** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11072?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11072?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#11071** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11071?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11072).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
I wasn't aware that the reserve method has guarantees. Just point out, `append_comments` method doesn't have any safety comment in [standard library](https://github.com/rust-lang/rust/blob/1b9efcd18f1cb95348c3ffadd4db47bfa15292e5/library/alloc/src/vec/mod.rs#L2586C1-L2596C6) as well 🤣

overlookmotel commented:
> I wasn't aware that the reserve method has guarantees. Just point out, `append_comments` method doesn't have any safety comment in [standard library](https://github.com/rust-lang/rust/blob/1b9efcd18f1cb95348c3ffadd4db47bfa15292e5/library/alloc/src/vec/mod.rs#L2586C1-L2596C6) as well 🤣

😱

overlookmotel commented:
### Merge activity

* **May 17, 6:15 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
docs(allocator/vec): correct safety comments for `RawVec::append_elements` (#11072)

Follow-on after #10884.

The original safety conditions aren't required any more, because of the change made to `Vec::reserve` in #10884 - `reserve` will panic if `other.len` or `self.len + other.len` exceeds `u32::MAX`. So nothing can go wrong if these conditions aren't satisfied.

Which is lucky, because `append` doesn't do anything to make sure `self.len + other.len <= u32::MAX`!

`append_elements` originally had no safety docs at all (naughty bumpalo!). So add some.
